src/device.rs

```diff
@@ -722,9 +729,66 @@ impl Device {
         }
     }
 
-    // TODO: clocking
+/// List time sources
+    pub fn list_time_sources(&self) -> Result<Vec<String>, Error> {
+        unsafe { string_list_result(|len_ptr| SoapySDRDevice_listTimeSources(self.ptr, len_ptr)) }
+    }
+
+    /// Get the current time source
+    pub fn get_time_source(&self) -> Result<String, Error> {
+        unsafe { string_result(SoapySDRDevice_getTimeSource(self.ptr)) }
+    }
+
+    /// Set the current time source
+    pub fn set_time_source<S: Into<Vec<u8>>>(&self, time_source: S) -> Result<(), Error> {
+        let time_source = CString::new(time_source).expect("Time source contained null");
+        unsafe {
+            SoapySDRDevice_setTimeSource(self.ptr, time_source.as_ptr());
+            check_error(())
+        }
+    }
+
+    /// Check whether there is a given hardware time source.
+    /// Hardware time sources are not the same as time sources (at least for UHD Devices)
+    /// UHD supported hw time sources: "PPS" or "" (i.e. None)
```

Comment:
In general I'd prefer not to have driver-specific details in this library; however since there's no API to list the supported time sources, and I think UHD is the only driver to support anything other than `None` here, I'll allow this for now.

Reply:
Seems reasonable.
Would you prefer to keep such comments out of the docs entirely or in a separate section?
For ease of use (from a new user perspective) it might be valuable to have some pointers for SDR-specific behaviour - with the usual drawbacks of these notes becoming out of sync with the rest of the library.

Reply:
I'm not sure what the best answer is. For anything that is not too rust-specific, it would be good to improve the documentation on the SoapySDR wiki or the individual driver repos, in hopes that upstream would keep them up to date.

Driver-specific examples/tests would be good as well, and serve as documentation for how to use those features from Rust.